contracts/governance/Staking/Staking.sol

```diff
@@ -774,4 +918,26 @@ contract Staking is
                 block.number;
         require(notSameBlock, "S20"); //S20 : "cannot be mined in the same block as last stake"
     }
+
+    /**
+     * @notice Max iteration for direct withdrawal from staking to prevent out of gas issue.
+     *
+     * @return max iteration value.
+     */
+    function getMaxVestingWithdrawIterations() public view returns (uint256) {
+        return maxVestingWithdrawIterations;
+    }
+
+    /**
+     * @dev set max withdraw iterations.
+     *
+     * @param maxIterations new max iterations value.
+     */
+    function setMaxVestingWithdrawIterations(uint256 maxIterations) external onlyAuthorized {
+        require(maxIterations > 0, "S21");
```

Comment:
pls replace the message with the full length text
this is a requirement for modularized Staking which will need to do at the next step of wrapping it into modules anyway

Reply:
Updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/d448c6258afce4aaede3f46f568b92fea9dc2e9d)